inclusive-design/wecount.inclusivedesign.ca
# feat: Create Resources page filter

## Description

This change introduces the Resources page's resource search & filter functionality, solving #437. Resources can be searched by search term and/or filtered by one or more of the following fields:
- Topic (determined by a resources' focus)
- Tags
- Media type

Searching by search term will search for resources which contain the term in one or more of the following fields:
- Title
- Abstract
- Media type
- Keywords
- Learn Tags
- Summary

Within a particular filter section, selecting more checkboxes will yield more results, as with a boolean inclusive "or". Selecting checkboxes between multiple filter sections will yield the intersection between each section's results, as with an "and".

## Steps to test

1. Go to the `/resources/` page: http://localhost:3000/resources/
2. The Topic filter section should be expanded by default, while the Tags and Media Types filter sections should be collapsed
3. Enter a search term into the search input text box
4. Press "Enter" to initiate the search
5. The page should reload and all three filter sections should now be expanded
6. Clicking the heading for each filter section should collapse it
7. The search results should include only results which contain the desired search term within one of the applicable fields. To verify this, open a couple of results to ensure the term is present on the page
8. Go back to the Resources page and click on a Topic checkbox
9. The page should reload and filter the resources to only those with the selected Topic
10. Click on another Topic checkbox
11. The page should reload and filter the resources to include only those within the selected Topics
12. Click on a tag, the box should indicate it is selected but nothing else should happen
13. Click "Apply Filters"
14. The page should reload and show only resources which are both in the selected Topics and contain the selected tag
15. Click on another tag, and then on "Apply Filters"
16. The page should reload and show resources which are in both selected Topics and which contain either of the two selected tags
17. Repeat the "Tags" steps with Media Types
18. Click on "Clear All Topics" in the Topic filter section
19. The page should reload, removing the filter by Topic and including results from all Topics
20. Click on "Clear All Tags" in the Tags filter section
21. The page should reload, removing the filter by Tags and including results regardless of Tags
22. Click on "Clear All Media Types" in the Media Types filter section
23. The page should reload, removing the filter by Media Types, as above
24. The results should now match the initial search with the search term
25. Remove the search term and press "Enter" to refresh the page
26. All resources should be listed with no filtering, matching the initial view from step 2
27. Click on several Tags and Media Types, then on a Topic, and let the page reload
28. Click on "Reset Filters"
29. The page should reload, removing all filters, again matching step 2


## Related issues

Resolves #437
